src/JustSaying/AwsTools/MessageHandling/Compression/IMessageBodyCompressor.cs

```diff
@@ -0,0 +1,7 @@
+namespace JustSaying.AwsTools.MessageHandling.Compression;
+
+internal interface IMessageBodyCompressor
+{
+    string ContentEncoding { get; }
+    string Compress(string messageBody);
```

Comment:
Can we make this async, or do we not have access to the required async methods the implementations use in all our TFMs?

Reply:
Fair point, I was thinking it should be sync because we are writing to and from in-memory streams. No strong opinion here from me.

Reply:
True, but if it's pluggable and we could make it async, it would enable async if for some reason it was being backed by a file or some other IO.